app/models/concerns/form526_rapid_ready_for_decision_concern.rb

```diff
@@ -118,6 +118,15 @@ def single_disability_eligible_for_mas?
     RapidReadyForDecision::Constants::MAS_DISABILITIES.include?(diagnostic_codes.first) && increase_only?
   end
 
+  # return whether all disabilities on this form are rated as not service-connected
+  def disabilities_not_service_connected?
+    disabilities.pluck('ratedDisabilityId').all? do |rated_id|
+      rated_id.present? && (all_rated_disabilities
+                              .find { |rated| rated_id == rated.rated_disability_id }
```

Comment:
If you wanted to optimize, you could call https://apidock.com/rails/Enumerable/index_by and return a hash for `all_rated_disabilities`. This way, you won't need a `find` and can use `all_rated_disabilities[rated_id]&.decision_code == 'NOTSVCCON'` instead, which is easier to interpret IMHO.